### PR TITLE
refactor: add function for deserializing and collecting packets - separate from channel receive

### PR DESCRIPTION
**Pull request by apfitzge**
#### Problem
Follow up to #27120. Working on completely separating deserialization from banking stage, it was convenient to have this function separated from the actual channel receiving for tests.

#### Summary of Changes
Create a separate function that deserializes and builds ReceivePacketResults from packet_batches and tracer stats.

Incremental step for #27009 

Fixes #

### CODE REVIEW

**Comment by apfitzge** [comment]
@Mergifyio rebase

**Review by carllin — APPROVED** [review]

File: core/src/packet_deserializer.rs [hunk 1]
```diff
@@ -45,26 +45,36 @@ impl PacketDeserializer {
...
         let (packet_batches, sigverify_tracer_stats_option) =
             self.receive_until(recv_timeout, capacity)?;
+        Ok(Self::deserialize_and_collect_packets(
+            &packet_batches,
+            sigverify_tracer_stats_option,
+        ))
+    }
 
+    /// Deserialize packet batches and collect them into ReceivePacketResults
+    fn deserialize_and_collect_packets(
```
Comment (taozhu-chicago) [inline core/src/packet_deserializer.rs:55]: nit: add tests for this function, as the PR's description suggests

**Review by taozhu-chicago — COMMENTED** [review]
l

File: core/src/packet_deserializer.rs (hunk 1 above)
Comment (apfitzge) [reply core/src/packet_deserializer.rs:55]: Ah, in the PR here I meant to make it easier to use in tests inside #27611.
I'll create another PR to add some testing here though.

**Pull request merged**